Source/levels/gendung.cpp

```diff
@@ -115,7 +115,7 @@ std::optional<Size> GetSizeForThemeRoom(int floor, Point origin, int minSize, in
 			}
 		}
 
-		int height = 0;
+		int height = i < room.width ? i : 0;
 		if (i < maxWidth) {

```

Comment:
The following can be done to avoid the remaining double scans but doing so will actually degrade performance (and readability).
```cpp
		if (i < maxWidth) {
			if (width > i)
				height++;
```